### DISCUSSION TITLE
Prisma migration broken again?

### DISCUSSION THREAD

**Discussion by tbenci** [prisma/prisma#24401] [Q&A] [answered]
### Question

I am running npx prisma migrate dev --name init and it is just hanging. Seems that was a problem a year ago.

I have proven I can connect to the database with my credentials through SQlectron.

{
    "dependencies": {
        "@clerk/nextjs": "^5.1.3",
        "@clerk/themes": "^2.1.8",
        "@prisma/client": "^5.14.0",
        "@radix-ui/react-dialog": "^1.0.5",
        "@radix-ui/react-label": "^2.0.2",
        "@radix-ui/react-separator": "^1.0.3",
        "@radix-ui/react-slot": "^1.0.2",
        "@reduxjs/toolkit": "^2.2.5",
        "@types/react-redux": "^7.1.33",
        "class-variance-authority": "^0.7.0",
        "clsx": "^2.1.1",
        "lucide-react": "^0.381.0",
        "next": "14.2.3",
        "next-themes": "^0.3.0",
        "prettier": "^3.2.5",
        "prettier-plugin-tailwindcss": "^0.6.1",
        "react": "^18",
        "react-dom": "^18",
        "react-icons": "^5.2.1",
        "react-redux": "^9.1.2",
        "react-toastify": "^10.0.5",
        "redux": "^5.0.1",
        "tailwind-merge": "^2.3.0",
        "tailwind-scrollbar": "^3.1.0",
        "tailwindcss-animate": "^1.0.7"
    },
    "devDependencies": {
        "@types/node": "^20",
        "@types/react": "^18",
        "@types/react-dom": "^18",
        "eslint": "^8",
        "eslint-config-next": "14.2.3",
        "postcss": "^8",
        "prisma": "^5.14.0",
        "tailwindcss": "^3.4.1",
        "typescript": "^5"
    },
    "name": "next-tailwind-clerk",
    "private": true,
    "scripts": {
        "build": "next build",
        "dev": "next dev --turbo",
        "lint": "next lint",
        "start": "next start"
    },
    "version": "0.1.0"
}

### How to reproduce (optional)

Just create a project, a schema and migrate

### Expected behavior (optional)

Should create the tables

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

// ------------------------------------------------------------------------------------
// COMPANY
// ------------------------------------------------------------------------------------

model Company {
  countryId  Int
  createdAt  DateTime @default(now())
  id         Int      @id @default(autoincrement())
  key        String   @default(uuid())
  name       String
  stateId    Int
  timeZoneId Int
  updatedAt  DateTime @updatedAt

  // Relationships

  authorisations Authorisation[]
  branches       Branch[]

  country  Country  @relation(fields: [countryId], references: [id])
  state    State    @relation(fields: [stateId], references: [id])
  TimeZone TimeZone @relation(fields: [timeZoneId], references: [id])
}

// ------------------------------------------------------------------------------------
// BRANCH
// ------------------------------------------------------------------------------------

model Branch {
  address                String
  companyId              Int
  contactTimeMessage     String
  countryId              Int
  createdAt              DateTime @default(now())
  email                  String
  externalProviderConfig Json?
  externalProviderCode   String?
  id                     Int      @id @default(autoincrement())
  key                    String   @unique @default(uuid())
  location               String
  name                   String
  stateId                Int
  telephone              String
  timeZoneId             Int
  updatedAt              DateTime @updatedAt

  // Relationships

  authorisations Authorisation[]

  company  Company  @relation(fields: [companyId], references: [id])
  country  Country  @relation(fields: [countryId], references: [id])
  state    State    @relation(fields: [stateId], references: [id])
  timeZone TimeZone @relation(fields: [timeZoneId], references: [id])
}

// ------------------------------------------------------------------------------------
// COUNTRY, STATE and TIMEZONE
// ------------------------------------------------------------------------------------

model Country {
  abbrev2     String
  abbrev3     String
  createdAt   DateTime @default(now())
  id          Int      @id @default(autoincrement())
  name        String
  numericCode String   @unique
  stateName   String
  updatedAt   DateTime @updatedAt

  // Relationships

  branches  Branch[]
  companies Company[]
  states    State[]
  timeZones TimeZone[]
}

model State {
  code       String
  countryId  Int
  createdAt  DateTime @default(now())
  id         Int      @id @default(autoincrement())
  nameLong   String
  nameShort  String
  timeZoneId Int
  updatedAt  DateTime @updatedAt

  // Relationships

  branches  Branch[]
  companies Company[]

  country  Country  @relation(fields: [countryId], references: [id])
  timeZone TimeZone @relation(fields: [timeZoneId], references: [id])
}

model TimeZone {
  countryId   Int
  createdAt   DateTime @default(now())
  description String
  id          Int      @id @default(autoincrement())
  updatedAt   DateTime @updatedAt
  zone        String

  // Relationships

  branches  Branch[]
  companies Company[]
  states    State[]

  country Country @relation(fields: [countryId], references: [id])
}

// ------------------------------------------------------------------------------------
// AUTHORISATION and USER
// ------------------------------------------------------------------------------------

enum AuthorisationLevel {
  ALL
  BRANCH
}

enum AuthorisationRole {
  ADMIN
  GUEST
  USER
}

model Authorisation {
  active    Boolean
  branchId  Int?
  companyId Int?
  createdAt DateTime           @default(now())
  id        Int                @id @default(autoincrement())
  level     AuthorisationLevel
  role      AuthorisationRole
  updatedAt DateTime           @updatedAt
  userId    Int

  // Relationships

  branch  Branch?  @relation(fields: [branchId], references: [id])
  company Company? @relation(fields: [companyId], references: [id])
  user    User     @relation(fields: [userId], references: [id])
}

model User {
  createdAt DateTime @default(now())
  email     String   @unique
  id        Int      @id @default(autoincrement())
  updatedAt DateTime @updatedAt

  // Relationships

  authorisations Authorisation[]
}
```

- Database: PostgreSQL - Supabase
- prisma: 5.14.0
- @prisma/client: 5.14.0
- Computed binaryTarget : windows
- Operating System : win32
- Architecture: x64
- Node.js: v21.4.0

**Comment by ludralph** [comment, marked as answer]
HI @tbenci 👋 

Thank you for reporting this.

Supabase needs you to also specify `DIRECT_URL` apart from `DATABASE_URL`.
```prisma
datasource db {
  provider  = "postgresql"
  url       = env("DATABASE_URL")
  directUrl = env("DIRECT_URL")
}
```
See also this [comment](https://github.com/prisma/prisma/issues/17338#issuecomment-2021842857) from the issue.

**Reply by tbenci** [reply]
Hi @ludralph 

I truly appreciate your response and it is now working (thank goodness :) ).

I was following my existing template app and had missed the DIRECT_URL (was unnecessary when I last did a migration it seems).

Just to confirm, I couldn't find specific instructions on what or where to get the DIRECT_URL but took the DATABASE_URL and changed the port of the DATABASE_URL to 5432.

Is that correct or have I missed something?

**Reply by ludralph** [reply]
Hi @tbenci 

DATABASE_URL="" # Set this to the Transaction connection pooler string
DIRECT_URL=""  # Set this to the Session connection pooler string 

You can find the information in the [Database settings](https://supabase.com/dashboard/project/_/settings/database) page of Supabase
